test/GenomeSet_basic_test.py

```diff
@@ -1,200 +1,217 @@
 """Basic GenomeSet tests."""
-from test.util import INFO_LENGTH
+from test.common_test import check_get_set, check_save_set_output, is_info_object
+from typing import Any
 
-from SetAPI.generic.constants import INC_ITEM_INFO, INC_ITEM_REF_PATHS
+import pytest
+from SetAPI.generic.constants import INC_ITEM_INFO, INC_ITEM_REF_PATHS, REF_PATH_TO_SET
+from SetAPI.genome.GenomeSetInterfaceV1 import GenomeSetInterfaceV1
 from SetAPI.SetAPIImpl import SetAPI
 
+API_CLASS = GenomeSetInterfaceV1
+SET_TYPE = "genome"
 
-def test_basic_save_and_get(
-    genome_refs: list[str],
+
+def save_genome_set(
     set_api_client: SetAPI,
     context: dict[str, str | list],
     ws_id: int,
-) -> None:
-    set_name = "set_of_genomes"
-    set_description = "my first genome set"
-
-    # create the set object
-    set_data = {
-        "description": set_description,
-        "items": [
-            {"ref": genome_refs[0], "label": "genome1"},
-            {"ref": genome_refs[1], "label": "genome2"},
-        ],
-    }
-    n_items = len(set_data["items"])
-
-    # test a save
-    res = set_api_client.save_genome_set_v1(
+    set_name: str,
+    set_data: dict[str, Any],
+    kwargs: None | dict[str, Any] = None,
+) -> dict[str, Any]:
+    """Given a set_name and set_data, save a genome set."""
+    if not kwargs:
+        kwargs = {}
+    return set_api_client.save_genome_set_v1(
         context,
         {
-            "data": set_data,
-            "output_object_name": set_name,
             "workspace_id": ws_id,
+            "output_object_name": set_name,
+            "data": set_data,
+            **kwargs,
```

Comment:
this one is slightly different as there's an additional argument added, so the args to save_set have to be changed slightly